arangod/IResearch/IResearchAqlAnalyzer.cpp

```diff
@@ -0,0 +1,562 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2020 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Andrei Lobov
+////////////////////////////////////////////////////////////////////////////////
+
+// otherwise define conflict between 3rdParty\date\include\date\date.h and
+// 3rdParty\iresearch\core\shared.hpp
+#if defined(_MSC_VER)
+#include "date/date.h"
+#endif
+
+#include "IResearchAqlAnalyzer.h"
+
+#include "utils/hash_utils.hpp"
+#include "utils/object_pool.hpp"
+
+#include "Aql/Ast.h"
+#include "Aql/AqlFunctionFeature.h"
+#include "Aql/AqlTransaction.h"
+#include "Aql/ExpressionContext.h"
+#include "Aql/Parser.h"
+#include "Aql/QueryString.h"
+#include "Basics/StaticStrings.h"
+#include "Basics/StringUtils.h"
+#include "Basics/VelocyPackHelper.h"
+#include "Basics/FunctionUtils.h"
+#include "IResearchCommon.h"
+#include "Logger/LogMacros.h"
+#include "VelocyPackHelper.h"
+#include "VocBase/vocbase.h"
+#include "RestServer/DatabaseFeature.h"
+#include "Transaction/SmartContext.h"
+#include "Utils/CollectionNameResolver.h"
+
+#include <Containers/HashSet.h>
+#include "VPackDeserializer/deserializer.h"
+#include <frozen/set.h>
+
+namespace {
+using namespace arangodb::velocypack::deserializer;
+using namespace arangodb::aql;
+
+constexpr const char QUERY_STRING_PARAM_NAME[] = "queryString";
```

Comment:
We typically place those in lib/basics/StaticStrings.h|.cpp

Happy that you use this :+1:

Reply:
It is "local use" string. I prefer to keep it local.